packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/analytics_client.dart

```diff
@@ -17,18 +16,32 @@ import 'package:meta/meta.dart';
 /// {@endtemplate}
 class AnalyticsClient {
   /// {@macro amplify_analytics_pinpoint_dart.analytics_client}
-  AnalyticsClient({
+  factory AnalyticsClient({
+    required AmplifyStorageScope storageScope,
     DeviceContextInfoProvider? deviceContextInfoProvider,
-    EndpointInfoStoreManager? endpointInfoStoreManager,
-  })  : _deviceContextInfoProvider = deviceContextInfoProvider,
-        _endpointInfoStoreManager = endpointInfoStoreManager ??
-            EndpointInfoStoreManager(
-              store: AmplifySecureStorageWorker(
-                config: AmplifySecureStorageConfig(
-                  scope: EndpointStorageScope.analyticsPinpoint.name,
-                ),
+    LegacyNativeDataProvider? legacyNativeDataProvider,
+    SecureStorageFactory? secureStorageFactory,
```

Comment:
Yes, AnalyticsClient will be a bit weird because it is used by Analytics and Push Notifications, and possibly more notification categories in the longer term future. 

Push Notifications will create AnalyticsClient with only storageScope and deviceContextInfoProvider defined.